text/0000-option-unwrap-unchecked.md

```diff
@@ -0,0 +1,73 @@
+- Feature Name: option_unwrap_unchecked
+- Start Date: 2015-4-28
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+This RFC adds a method to the `Option` enum which allows it to be
+unsafely unwrapped without checking for `None`.
+
+# Motivation
+
+In cases where a developer can determine that an `Option` is `Some(v)`
+but the compiler cannot, this method allows a developer to unsafely
+opt-in to an optimization which will avoid a conditional check. In my
+own uses I find cases where I know that an `Option` is `Some(v)`. I
+present a case from the `libcollections linked_list.rs`:
+
+    fn next(&mut self) -> Option<&'a A> {
+        if self.nelem == 0 {
+            return None;
+        }
+        self.head.as_ref().map(|head| {
```

Comment:
The first `if` is necessary because the iterator is double ended. Without checking it, next will only return `None` at the tail. The argument against `unwrap()` is about the fact that it panics (as far as I can tell). This is not a concern for this proposal. 

I'd be curious to see your recommendation as to how we modify the `linked_list` and its iterator to consolidate the semi-related datums. Otherwise, I see `unwrap_unchecked` or an alternative as the right way to do it.